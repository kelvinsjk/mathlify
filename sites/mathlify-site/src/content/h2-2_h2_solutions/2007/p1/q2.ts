import { Answer } from '$lib/classes/answer';
import { QED } from '$lib/utils/typesetting/utils';
import { Polynomial } from 'mathlify';
import { generateDomain } from '$content/h2-1_h2_learn/01_functions/02_inverse/02_domain/02_practice';
import { improperInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { compositeExists } from '$content/h2-1_h2_learn/01_functions/03_composite/01_existence/02_practice';
import { compositeFormula } from '$content/h2-1_h2_learn/01_functions/03_composite/02_formula/02_practice';

export const answer = new Answer();

// ! Complete
// f(x) = 1/(x-3)
// g(x) = x^2

const [a, b, c] = [-3, 0, 1];
const f = new Polynomial([1, a]).reciprocal();
const g = Polynomial.ofDegree(2);
const fState: Parameters<typeof improperInverse>[0] = {
	fnType: 'frac',
	a,
	b,
	c,
	restriction: false,
	unknownConstants: false,
	definition: false
};
const gState = {
	fnType: 'quadratic',
	a: 0,
	b: 0,
	c: 0,
	restriction: false,
	unknownConstants: false
} as const;
const compositeStateFg: Parameters<typeof compositeExists>[0] = {
	f: fState,
	g: gState,
	fg: true
};

// a
{
	const { ans: ans2, soln: soln2 } = compositeExists(compositeStateFg, [f, g]);
	const { ans, soln } = compositeFormula(
		[f, generateDomain(fState), g, generateDomain(gState)],
		false,
		true,
		{ ansInline: true, QED: true }
	);
	answer.addPart(ans + '\\\n' + ans2, soln + '\n\n' + soln2);
}
// b
{
	const { ans, soln } = improperInverse(fState, {
		definition: false,
		rhs: f,
		QED,
		omitReasoningInAns: true
	});
	answer.addPart(ans, soln);
}
