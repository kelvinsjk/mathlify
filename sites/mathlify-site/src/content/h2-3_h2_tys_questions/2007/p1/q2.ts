/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Polynomial } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

//f(x) = 1/(x-3)
//g(x) = x^2

const a = -3;
const f = new Polynomial([1, a]).reciprocal();
const g = Polynomial.ofDegree(2);

question.addBody(
	mathlifier`Functions ${'f'}
and ${'g'}
are defined by
$${'align*'}
&f: x \\mapsto ${f}  \\quad && \\text{for } x \\in \\mathbb{R}, x \\neq 3, \\\\
&g: x \\mapsto ${g}  \\quad && \\text{for } x \\in \\mathbb{R}.
`
);
// a
{
	question.addPart(
		mathlifier`Only one of the
composite functions ${'fg'}
and ${'gf'}
exists. Give a definition (including the domain)
of the composite that exists, and explain why
the other composite does not exist.`,
		{ marks: 3 }
	);
}
// b
{
	question.addPart(
		mathlifier`Find ${'f^{-1}(x)'}
and state the domain of ${'f^{-1}'}.`,
		{ marks: 3 }
	);
}
