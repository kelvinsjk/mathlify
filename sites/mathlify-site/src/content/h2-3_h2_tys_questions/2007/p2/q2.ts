/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

question.addBody(mathlifier`Out of syllabus (method of differences).`, { marks: 11 });
