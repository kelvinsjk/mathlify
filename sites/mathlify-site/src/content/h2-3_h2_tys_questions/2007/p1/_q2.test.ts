import { expect, it } from 'vitest';
import { question } from './q2';

it('2007p1q2 qn', () => {
	expect(question).toMatchSnapshot();
});
