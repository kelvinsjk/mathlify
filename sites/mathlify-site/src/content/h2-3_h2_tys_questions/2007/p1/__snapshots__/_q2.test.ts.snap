// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2007p1q2 qn 1`] = `
Question {
  "question": {
    "body": "Functions $\` f \`
and $\` g \`
are defined by
$$\` \\begin{align*}
&f: x \\mapsto \\frac{1}{x - 3}  \\quad && \\text{for } x \\in \\mathbb{R}, x \\neq 3, \\\\
&g: x \\mapsto x^2  \\quad && \\text{for } x \\in \\mathbb{R}.
\\end{align*} \`",
    "parts": [
      {
        "body": "Only one of the
composite functions $\` fg \`
and $\` gf \`
exists. Give a definition (including the domain)
of the composite that exists, and explain why
the other composite does not exist.",
        "marks": 3,
      },
      {
        "body": "Find $\` f^{-1}(x) \`
and state the domain of $\` f^{-1}. \`",
        "marks": 3,
      },
    ],
  },
}
`;
